t/testdata/property/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func main() {
+	proxywasm.SetVMContext(&vmContext{})
+}
+
+type vmContext struct {
+	// Embed the default VM context here,
+	// so that we don't need to reimplement all the methods.
+	types.DefaultVMContext
+}
+
+// Override types.DefaultVMContext.
+func (*vmContext) NewPluginContext(contextID uint32) types.PluginContext {
+	return &pluginContext{}
+}
+
+type pluginContext struct {
+	// Embed the default plugin context here,
+	// so that we don't need to reimplement all the methods.
+	types.DefaultPluginContext
+}
+
+// Override types.DefaultPluginContext.
+func (*pluginContext) NewHttpContext(contextID uint32) types.HttpContext {
+	return &httpLifecycle{contextID: contextID}
+}
+
+func (*pluginContext) OnPluginStart(pluginConfigurationSize int) types.OnPluginStartStatus {
+	return true
+}
+
+type httpLifecycle struct {
+	// Embed the default http context here,
+	// so that we don't need to reimplement all the methods.
+	types.DefaultHttpContext
+	contextID uint32
+}
+
+var variables = []string{"arg_test", "scheme", "host", "request_uri", "uri"}
```

Comment:
Better to pass the test data via configuration. You can take a look at https://github.com/api7/wasm-nginx-module/pull/61

Reply:
I migrated it to a table driven test implemented using test scripts.